src/wp-includes/class-wpdb.php

```diff
@@ -1553,106 +1555,130 @@ public function prepare( $query, ...$args ) {
-					 * and we ensure string escaping is used as a safe default (e.g. even if 'x').
-					 */
-					$argnum_pos = strpos( $format, '$' );
+				$argnum_pos = strpos( $format, '$' );
+				if ( false !== $argnum_pos ) {
+					$argnum_value = (int) substr( $format, 0, $argnum_pos );
+					$format       = substr( $format, ( $argnum_pos + 1 ) );
+				} else {
+					$argnum_value = ++$arg_current; // Argnum starts at 1.
+				}
+
+				$new_argnum = ( $argnum_value + $arg_offset );
 
-					if ( false !== $argnum_pos ) {
-						$arg_strings[] = ( ( (int) substr( $format, 0, $argnum_pos ) ) - 1 );
+				if ( $variadic ) {
+					if ( 'i' === $type ) {
+						$new_prefix        = '`%';
+						$new_suffix        = $format . 's`';
+						$arg_identifiers[] = ( $argnum_value - 1 );
+					} elseif ( 'd' === $type || 'F' === $type ) {
+						$new_prefix = '%'; // No need to quote integers or floats.
+						$new_suffix = $format . $type;
 					} else {
-						$arg_strings[] = $arg_id;
+						$new_prefix = "'%";
+						$new_suffix = $format . $type . "'";
 					}
-
-					/*
-					 * Unquoted strings for backward compatibility (dangerous).
-					 * First, "numbered or formatted string placeholders (eg, %1$s, %5s)".
-					 * Second, if "%s" has a "%" before it, even if it's unrelated (e.g. "LIKE '%%%s%%'").
-					 */
-					if ( true !== $this->allow_unsafe_unquoted_parameters
-						/*
-						 * Note: str_ends_with() is not used here, as this file can be included
-						 * directly outside of WordPress core, e.g. by HyperDB, in which case
-						 * the polyfills from wp-includes/compat.php are not loaded.
-						 */
-						|| ( '' === $format && '%' !== substr( $split_query[ $key - 1 ], -1, 1 ) )
-					) {
-						$placeholder = "'%" . $format . "s'";
+					$new_placeholder = '';
+					$arg_count       = count( $args[ ( $argnum_value - 1 ) ] ); // The argnum in $format starts from 1, but $args index start from 0.
+					for ( $k = 0; $k < $arg_count; $k++ ) {
+						$new_placeholder .= $new_prefix . ( $new_argnum + $k ) . '$' . $new_suffix . ',';
+					}
+					$new_placeholder = substr( $new_placeholder, 0, -1 );
+					$arg_offset     += ( $arg_count - 1 ); // The arg already counts as 1, the offset is how many extra to move.
+					$arg_variadics[] = ( $argnum_value - 1 );
+				} elseif ( 'i' === $type ) {
+					$new_placeholder   = '`%' . $new_argnum . '$' . $format . 's`';
+					$arg_identifiers[] = ( $argnum_value - 1 );
+				} elseif ( 'd' === $type || 'F' === $type ) {
+					$new_placeholder = '%' . $new_argnum . '$' . $format . $type; // No need to quote integers or floats.
+				} else { // i.e. ( 's' === $type ).
+					if ( true === $this->allow_unsafe_unquoted_parameters && ( $set_format || null !== $escaped ) ) {
+						// Unquoted strings for backward compatibility (dangerous).
+						// First, "numbered or formatted string placeholders (eg, %1$s, %5s)"
+						// Second, if "%s" has a "%" before it, even if it's unrelated (e.g. "LIKE '%%%s%%'").
```

Comment:
```suggestion
						/*
						 * Unquoted strings for backward compatibility (dangerous).
						 * First, "numbered or formatted string placeholders (eg, %1$s, %5s)"
						 * Second, if "%s" has a "%" before it, even if it's unrelated (e.g. "LIKE '%%%s%%'").
						 */
```
Per the [PHP Inline Documentation Standards](https://developer.wordpress.org/coding-standards/inline-documentation-standards/php/#5-2-multi-line-comments), this should use the multiline comment format.